subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/MavenArtifactRepository.java

```diff
@@ -127,5 +127,26 @@
         void artifact();
     }
 
+    /**
+     * The different common types of Maven repositories.
+     *
+     * @since 5.1
+     */
+    @Incubating
+    enum RepositoryKind {
+        RELEASES_ONLY,
+        SNAPSHOTS_ONLY,
+        MIXED
+    }
 
+    /**
+     * Sets the repository kind. By default, Gradle assumes
+     * a Maven repository may contain both snapshots and releases.
+     *
+     * @param kind the repository kind
+     *
+     * @since 5.1
+     */
+    @Incubating
+    void setKind(RepositoryKind kind);
```

Comment:
How about two booleans, since the two are basically orthogonal? ```
maven {
   url = "..."
   releases = true
   snapshots = false
}
```

Reply:
Agree, and by default they'd both be `true`?  Also makes the DSL a little cleaner.  👍

Reply:
There are 2 reasons for me:
- it's 2 options to configure a single thing, which, honestly, I don't find super intuitive
- the "repository kind" may evolve as we start publishing, for example, native libraries to Maven repositories. The repository kind could become more evolved than just a couple of flags.